### PR TITLE
Feature/idp 530 create pipeline

### PR DESCRIPTION
**Pull request by decarufe**
## Context
As an IDP developer I want out event grid emulator project to have pipelines so we don’t have to do things manually.

We want to have our usual CI pipeline that will make sure the project can be built and all the tests pass. We also want to make sure we can build the docker image without publishing it.

We’ll also want to have a CD pipeline that run whenever a branch is merged back into main. This pipeline should build a new docker image and publish it on the workleap dockerhub organisation.

We need to have a semgrep pipeline.

NB: Not sure if the workleap dockerhub organisation has been created or not, so this might be a required step to do.

## What have been done
CI/CD and semgrep pipelines are created

CD pipeline publishes a docker image on dockerhub

### CODE REVIEW

File: src/EventGridEmulator/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,11 @@
 ﻿FROM mcr.microsoft.com/dotnet/aspnet:7.0-alpine AS base
 WORKDIR /app
-EXPOSE 8080
+EXPOSE 5000
```
Comment (Gcaya) [inline src/EventGridEmulator/Dockerfile:3]: I think I'd keep it at 8080, I feel it's the usual standard when you can't use the privileged port 80. As of  .net 8 images the default port is not 8080, could be good to follow this aswell. https://learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port

File: src/EventGridEmulator/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 ENV DOTNET_CLI_TELEMETRY_OPTOUT=true
 ENV DOTNET_ENVIRONMENT=Production
-ENV ASPNETCORE_URLS=http://+:8080
+ENV ASPNETCORE_URLS=http://+:5000
```
Comment (Gcaya) [inline src/EventGridEmulator/Dockerfile:6]: Same comment as the one above!

File: src/EventGridEmulator/Dockerfile [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+ENV DOCKER_HOST=tcp://localhost:2375
+ENV DOCKER_DRIVER=overlay
```
Comment (Gcaya) [inline src/EventGridEmulator/Dockerfile:8]: Are those two necessary?

File: src/HelloPublisher/Program.cs [hunk 4]
```diff
@@ -1,17 +1,19 @@
-﻿using Azure;
+using Azure;
 using Azure.Identity;
 using Azure.Messaging.EventGrid;
 
 var client1 = new EventGridPublisherClient(
```
Comment (Gcaya) [inline src/HelloPublisher/Program.cs:5]: If we're to keep only 1 client I'd call it eventGridClient, what do you think?

File: src/HelloPublisher/Program.cs [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
-
-var client2 = new EventGridPublisherClient(
-    new Uri("https://localhost:5000/activities-eg/api/events"),
+    new Uri("https://localhost:6000/webhook-200"),
     new AzureCliCredential());
 
+// var client2 = new EventGridPublisherClient(
+//    new Uri("https://localhost:6000/activities-eg/api/events"),
+//    new AzureCliCredential());
+
```
Comment (Gcaya) [inline src/HelloPublisher/Program.cs:12]: Probably remove this?

File: src/HelloPublisher/Program.cs [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
 var data = new Dictionary<string, string>();
 data["foo"] = "bar";
 
-client1.SendEvent(new EventGridEvent("thesubject", "theeventtype", "1.0", data));
-client2.SendEvent(new EventGridEvent("thesubject", "theeventtype", "1.0", data));
+Response response = client1.SendEvent(new EventGridEvent("thesubject", "theeventtype", "1.0", data));
+Console.WriteLine(response.Status);
+Console.ReadLine();
+// client2.SendEvent(new EventGridEvent("thesubject", "theeventtype", "1.0", data));
```
Comment (Gcaya) [inline src/HelloPublisher/Program.cs:19]: Remove this too?

File: src/HelloPublisher/Program.cs [hunk 7]
```diff
@@ -1,17 +1,19 @@
-﻿using Azure;
+using Azure;
```
Comment (Gcaya) [inline src/HelloPublisher/Program.cs:1]: Wait this file is not meant to be part of the source code right?

**Review by Gcaya — DISMISSED** [review]

File: src/EventGridEmulator/Dockerfile (hunk 3 above)
Comment (decarufe) [reply src/EventGridEmulator/Dockerfile:8]: Maybe something I don't understand but it wasn't working without that.

Comment (decarufe) [reply src/EventGridEmulator/Dockerfile:8]: I didn't really tought too much about the code. I focused on the pipeline. Should we create a task to revisit the code?

File: src/HelloPublisher/Program.cs (hunk 7 above)
Comment (decarufe) [reply src/HelloPublisher/Program.cs:1]: I didn't really tought too much about the code. I focused on the pipeline. Should we create a task to revisit the code?

File: src/HelloPublisher/Program.cs (hunk 5 above)
Comment (decarufe) [reply src/HelloPublisher/Program.cs:12]: Agreed.

File: src/HelloPublisher/Program.cs (hunk 6 above)
Comment (decarufe) [reply src/HelloPublisher/Program.cs:19]: Agreed too.

File: src/EventGridEmulator/Dockerfile (hunk 3 above)
Comment (decarufe) [reply src/EventGridEmulator/Dockerfile:8]: Do you know how to make it work without it?

File: src/HelloPublisher/Program.cs (hunk 7 above)
Comment (Gcaya) [reply src/HelloPublisher/Program.cs:1]: Yeah we could have a story to just prettify what's needed

**Review by Gcaya — APPROVED** [review]

**Pull request merged**